### PR TITLE
Add packages architecture future

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR declares the future architecture I think this repo should follow. In this proposition, I create different packages than the ones that current exists and I try to explain a little bit their responsibility and what they contain.

I hope that you, dear reviewer, comment on each missing aspect of this, always evolving, future.

## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
<em>Describe the steps with bullet points. Is there any external link that can be used to better test it or an example?</em> 

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Packages
+
+All packages follow Gatsby's [plugin naming scheme](https://www.gatsbyjs.com/docs/plugins-themes-and-starters/). 
+
+* **gatsby-plugin-graphql** adds Webpack and Babel configuration so we can share queries and types between Gatsby and any other GraphQL server. This plugin is essential for dynamic, client side queries
+
+* **gatsby-plugin-i18n** is an optimized react-intl capable plugin. This plugin pre-optimizes messages we can serve react-intl without an ICU parser to the browser. Also, this plugins adds all necessary providers to the React tree
```
Comment (emersonlaurentino) [inline docs/architecture.md:7]:
```suggestion
* **gatsby-plugin-i18n** is an optimized react-intl capable plugin. This plugin pre-optimizes messages we can serve react-intl without an ICU parser to the browser. Also, these plugins adds all necessary providers to the React tree.
```

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-theme-ui** is an optimized theme-ui plugin for Gatsby. It includes many optmizations, not only on `theme-ui` but also on `emotion`. The goal is to merge this package into their community based versions so all the community benefits from such optimizations
```
Comment (emersonlaurentino) [inline docs/architecture.md:9]:
```suggestion
* **gatsby-plugin-theme-ui** is an optimized theme-ui plugin for Gatsby. It includes many optimizations, not only on `theme-ui` but also on `emotion`. The goal is to merge this package into their community-based versions so all the community benefits from such optimizations.
```

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-nginx** allows you to serve your Gatsby site using `nginx`. This plugin creates a Netlify like experience where all dynamic pages, redirects and reverse proxies are merged automatically into a `nginx.conf` file. Just pick this file and start serving your site from anywhere
```
Comment (emersonlaurentino) [inline docs/architecture.md:11]:
```suggestion
* **gatsby-plugin-nginx** allows you to serve your Gatsby site using `nginx`. This plugin creates a Netlify like experience where all dynamic pages, redirects, and reverse proxies are merged automatically into an `nginx.conf` file. Just pick this file and start serving your site from anywhere.
```

File: docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-source-vtex** connects the Gatsby data layer to VTEX's GraphQL graph
```
Comment (emersonlaurentino) [inline docs/architecture.md:13]:
```suggestion
* **gatsby-source-vtex** connects the Gatsby data layer to VTEX's GraphQL.
```

File: docs/architecture.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-theme-store** contains a Web SDK and is the main skeleton of an ecommerce. In it you will find the web's best practices, like above/below the fold techniques an first class Pixel integration. Note that this repo should NOT contain any non structurant component
```
Comment (emersonlaurentino) [inline docs/architecture.md:15]:
```suggestion
* **gatsby-theme-store** contains a Web SDK and is the main skeleton of an ecommerce. In it, you will find the web's best practices, like above/below the fold techniques and first-class Pixel integration. Note that this repo should NOT contain any nonstructural component.
```

File: docs/architecture.md (hunk 1 above)
Comment (klzns) [inline docs/architecture.md:7]: This doesn't mean that dynamic interpolations would not happen right?

I mean, if I have a component that renders something like this based on the cart number of items:

> There are 3 products in your cart.

or

> There are 1 product in your cart.

or

> There are no products in your cart.

This should be handled by an i18n library in the client

File: docs/architecture.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-checkout** implements checkout's Web SDK. It can be used without `gatsby-theme-store` to create extremelly custom experiences
```
Comment (emersonlaurentino) [inline docs/architecture.md:17]:
```suggestion
* **gatsby-plugin-checkout** implements checkout's Web SDK. It can be used without `gatsby-theme-store` to create extremely custom experiences.
```

File: docs/architecture.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-vtex-cms** integrates VTEX CMS with Gatsby. Live preview and pilot your ecommerce directly from VTEX Admin. Note that this overwrites any page in your gatsby project
```
Comment (emersonlaurentino) [inline docs/architecture.md:19]:
```suggestion
* **gatsby-plugin-vtex-cms** integrates VTEX CMS with Gatsby. Live preview and pilot your ecommerce directly from VTEX Admin. Note that this overwrites any page in your gatsby project.
```

File: docs/architecture.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+* **lighthouse-config** contains an ecommerce focused Lighthouse CI configuration
```
Comment (emersonlaurentino) [inline docs/architecture.md:21]:
```suggestion
* **lighthouse-config** contains an ecommerce focused Lighthouse CI configuration.
```

File: docs/architecture.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+* **store-ui** contains many building blocks for creating your store. All components follows [this pattern](/components.md). Use it's storybook to find the right component for your ecommerce experience
```
Comment (emersonlaurentino) [inline docs/architecture.md:23]:
```suggestion
* **store-ui** contains many building blocks for creating your store. All components follows [this pattern](/components.md). Use it's storybook to find the right component for your ecommerce experience.
```

File: docs/architecture.md (hunk 5 above)
Comment (klzns) [inline docs/architecture.md:15]: Not clear what is the API here. Maybe small pieces would be easier to maintain and do breaking changes.

File: docs/architecture.md (hunk 6 above)
Comment (klzns) [inline docs/architecture.md:17]: Is that related to VTEX's Checkout?

File: docs/architecture.md (hunk 1 above)
Comment (tlgimenes) [reply docs/architecture.md:7]: No, we still have react-intl, however, we implement [this guide](https://formatjs.io/docs/guides/advanced-usage#pre-compiling-messages) to reduce react-intl sizes and blocking. time

Comment (tlgimenes) [inline docs/architecture.md:7]:
```suggestion
* **gatsby-plugin-i18n** is an optimized react-intl capable plugin. This plugin pre-optimizes messages we can serve react-intl without an ICU parser to the browser. Also, this plugin adds all necessary providers to the React tree
```

File: docs/architecture.md (hunk 6 above)
Comment (tlgimenes) [reply docs/architecture.md]: Can be ! The idea is that this plugin will have all orderForm related APIs and maybe the checkout flow. Maybe it's better for this to be **gatsby-theme-checkout**, what do you think ?

File: docs/architecture.md (hunk 5 above)
Comment (tlgimenes) [reply docs/architecture.md]: This package should contain basically `/sdk` `/pages` `/templates` and some structuring components in `/components`, like BelowTheFold.tsx and AboveTheFold.tsx from `gatsby-theme-vtex`

The `/sdk` should contain all basic store logic, like search pagination, sales channel, auth, queries, etc.
The `/pages` and `/templates` are the webpack entrypoints
The `/components` have  components that MUST be shadowed, like Layout, BelowTheFold.tsx and AboveTheFold.tsx

File: docs/architecture.md (hunk 1 above)
Comment (klzns) [reply docs/architecture.md]: Awesome!

File: docs/architecture.md (hunk 6 above)
Comment (klzns) [reply docs/architecture.md]: Shouldn't those package have vtex in its name? gatsby-theme-vtex-checkout

File: docs/architecture.md (hunk 5 above)
Comment (klzns) [reply docs/architecture.md]: Does it make sense to split the sdk packages? I'm imagining a situation where we want to do a breaking change of a specific SDK, the user wouldn't need to update the basic plugin.

Also **gatsby-theme-vtex-store** isn't a better name?

File: docs/architecture.md (hunk 6 above)
Comment (tlgimenes) [reply docs/architecture.md]: I don't think so, do you know about the "OpenStore" project ?

File: docs/architecture.md (hunk 5 above)
Comment (tlgimenes) [reply docs/architecture.md]: I'm really not sure about this one. The tradeoff I see in here is that splitting the packages give you greater flexibility, however maintaining a monolithic structure is easier for new people joining and makes us keep the codebase small. However, I really don't know.

About the package naming scheme, I think these packages can be the beginning of the OpenStore initiative

File: docs/architecture.md (hunk 6 above)
Comment (emersonlaurentino) [reply docs/architecture.md]: I like the suggestion of @klzns 

> gatsby-theme-vtex-checkout

Comment (tlgimenes) [reply docs/architecture.md]: Also, there is something I forgot to say. All packages are already scoped by the `@vtex` namespace. So the full name would be: `@vtex/gatsby-theme-checkout`. That's why I think we don't need the `-vtex-` in the middle of the name, like in `@vtex/gatsbty-theme-vtex-checkout`

Comment (klzns) [reply docs/architecture.md]: oh, okk, that's make sense!

File: docs/architecture.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
+# Packages
+
+All packages follow Gatsby's [plugin naming scheme](https://www.gatsbyjs.com/docs/plugins-themes-and-starters/). 
+
+* **gatsby-plugin-graphql** adds Webpack and Babel configuration so we can share queries and types between Gatsby and any other GraphQL server. This plugin is essential for dynamic, client-side queries.
```
Comment (klzns) [inline docs/architecture.md:5]:
```suggestion
* **@vtex/gatsby-plugin-graphql** adds Webpack and Babel configuration so we can share queries and types between Gatsby and any other GraphQL server. This plugin is essential for dynamic, client-side queries.
```

File: docs/architecture.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-i18n** is an optimized react-intl capable plugin. This plugin pre-optimizes messages we can serve react-intl without an ICU parser to the browser. Also, this plugin adds all necessary providers to the React tree
```
Comment (klzns) [inline docs/architecture.md:7]:
```suggestion
* **@vtex/gatsby-plugin-i18n** is an optimized react-intl capable plugin. This plugin pre-optimizes messages we can serve react-intl without an ICU parser to the browser. Also, this plugin adds all necessary providers to the React tree
```

File: docs/architecture.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-theme-ui** is an optimized theme-ui plugin for Gatsby. It includes many optimizations, not only on `theme-ui` but also on `emotion`. The goal is to merge this package into their community-based versions so all the community benefits from such optimizations.
```
Comment (klzns) [inline docs/architecture.md:9]:
```suggestion
* **@vtex/gatsby-plugin-theme-ui** is an optimized theme-ui plugin for Gatsby. It includes many optimizations, not only on `theme-ui` but also on `emotion`. The goal is to merge this package into their community-based versions so all the community benefits from such optimizations.
```

File: docs/architecture.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-nginx** allows you to serve your Gatsby site using `nginx`. This plugin creates a Netlify like experience where all dynamic pages, redirects, and reverse proxies are merged automatically into an `nginx.conf` file. Just pick this file and start serving your site from anywhere.
```
Comment (klzns) [inline docs/architecture.md:11]:
```suggestion
* **@vtex/gatsby-plugin-nginx** allows you to serve your Gatsby site using `nginx`. This plugin creates a Netlify like experience where all dynamic pages, redirects, and reverse proxies are merged automatically into an `nginx.conf` file. Just pick this file and start serving your site from anywhere.
```

File: docs/architecture.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-source-vtex** connects the Gatsby data layer to VTEX's GraphQL.
```
Comment (klzns) [inline docs/architecture.md:13]:
```suggestion
* **@vtex/gatsby-source-vtex** connects the Gatsby data layer to VTEX's GraphQL.
```

File: docs/architecture.md [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-theme-store** contains a Web SDK and is the main skeleton of an ecommerce. In it, you will find the web's best practices, like above/below the fold techniques and first-class Pixel integration. Note that this repo should NOT contain any nonstructural component.
```
Comment (klzns) [inline docs/architecture.md:15]:
```suggestion
* **@vtex/gatsby-theme-store** contains a Web SDK and is the main skeleton of an ecommerce. In it, you will find the web's best practices, like above/below the fold techniques and first-class Pixel integration. Note that this repo should NOT contain any nonstructural component.
```

File: docs/architecture.md [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-checkout** implements checkout's Web SDK. It can be used without `gatsby-theme-store` to create extremely custom experiences.
```
Comment (klzns) [inline docs/architecture.md:17]:
```suggestion
* **@vtex/gatsby-plugin-checkout** implements checkout's Web SDK. It can be used without `gatsby-theme-store` to create extremely custom experiences.
```

File: docs/architecture.md [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+
+* **gatsby-plugin-vtex-cms** integrates VTEX CMS with Gatsby. Live preview and pilot your ecommerce directly from VTEX Admin. Note that this overwrites any page in your gatsby project.
```
Comment (klzns) [inline docs/architecture.md:19]:
```suggestion
* **@vtex/gatsby-plugin-vtex-cms** integrates VTEX CMS with Gatsby. Live preview and pilot your ecommerce directly from VTEX Admin. Note that this overwrites any page in your gatsby project.
```

File: docs/architecture.md [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+* **lighthouse-config** contains an ecommerce focused Lighthouse CI configuration.
```
Comment (klzns) [inline docs/architecture.md:21]:
```suggestion
* **@vtex/lighthouse-config** contains an ecommerce focused Lighthouse CI configuration.
```

File: docs/architecture.md [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+
+* **store-ui** contains many building blocks for creating your store. All components follows [this pattern](/components.md). Use it's storybook to find the right component for your ecommerce experience.
```
Comment (klzns) [inline docs/architecture.md:23]:
```suggestion
* **@vtex/store-ui** contains many building blocks for creating your store. All components follows [this pattern](/components.md). Use it's storybook to find the right component for your ecommerce experience.
```

**Review by klzns — APPROVED** [review]

File: docs/architecture.md [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+* **@vtex/gatsby-plugin-nginx** allows you to serve your Gatsby site using `nginx`. This plugin creates a Netlify like experience where all dynamic pages, redirects, and reverse proxies are merged automatically into an `nginx.conf` file. Just pick this file and start serving your site from anywhere.
+
+* **@vtex/gatsby-source-vtex** connects the Gatsby data layer to VTEX's GraphQL.
+
+* **@vtex/gatsby-theme-store** contains a Web SDK and is the main skeleton of an ecommerce. In it, you will find the web's best practices, like above/below the fold techniques and first-class Pixel integration. Note that this repo should NOT contain any nonstructural component.
+
+* **@vtex/gatsby-plugin-checkout** implements checkout's Web SDK. It can be used without `gatsby-theme-store` to create extremely custom experiences.
+
+* **@vtex/gatsby-plugin-vtex-cms** integrates VTEX CMS with Gatsby. Live preview and pilot your ecommerce directly from VTEX Admin. Note that this overwrites any page in your gatsby project.
```
Comment (emersonlaurentino) [inline docs/architecture.md:19]: Doesn't using the word `vtex` leave the plugin name redundant since it already has the **@vtex** prefix?

```suggestion
* **@vtex/gatsby-plugin-cms** integrates VTEX CMS with Gatsby. Live preview and pilot your ecommerce directly from VTEX Admin. Note that this overwrites any page in your gatsby project.
```
